### PR TITLE
Patch MockExecutor in tests instead of "prod" code

### PR DESCRIPTION
**Pull request by o-nikolas**
tl;dr: Patch the MockExecutor into the executor list within test code rather than core/prod code.

Full story: The MockExecutor, which is used for testing, was added to the list of executors in "production"/core code in #27941
This can lead to that mock executor being loaded at runtime once Airflow is installed (when the `tests/test_utils/...` code is no longer present) and the import/load fails (new code I'm adding in a separate PR hits this case). So patch within test code instead of prod code.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by pierrejeambrun — APPROVED** [review]
Good catch. The dynamic import for `MockExecutor` wasn't actually resolve in production code (yet), but this was a ticking bomb indeed.

This is much cleaner thanks.

**Review by ferruzzi — APPROVED** [review]
Nice and clean :+1:

**Review by vincbeck — APPROVED** [review]
